subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/DefaultTaskCacheabilityResolver.java

```diff
@@ -59,7 +59,7 @@ public DefaultTaskCacheabilityResolver(RelativeFilePathResolver relativeFilePath
         if (overlappingOutputs != null) {
             String relativePath = relativeFilePathResolver.resolveForDisplay(overlappingOutputs.getOverlappedFilePath());
             return Optional.of(new CachingDisabledReason(CachingDisabledReasonCategory.OVERLAPPING_OUTPUTS,
-                "Gradle does not know how file '" + relativePath + "' was created (output property '" + overlappingOutputs.getPropertyName() + "'). Task output caching requires exclusive access to output paths to guarantee correctness."));
+                "Gradle does not know how file '" + relativePath + "' was created (output property '" + overlappingOutputs.getPropertyName() + "'). Task output caching requires exclusive access to output paths to guarantee correctness (each path only declared as the output of one task, transform, etc)."));
```

Comment:
Transforms have no ability to declare their outputs (and something similar is among the plans for tasks, but I digress), so let's not mention them here. This is also talking about task outputs here.

```suggestion
                "Gradle does not know how file '" + relativePath + "' was created (output property '" + overlappingOutputs.getPropertyName() + "'). Task output caching requires exclusive access to output paths to guarantee correctness (i.e. multiple tasks are not allowed to produce output in the same location)."));
```

(Yes, in theory you could be really evil and intentionally point your task to create something in `$buildDir/.transforms/...` to overlap with the output of a transform, but that is not something we want to defend against here.)